### ISSUE TITLE
[Please Help] Click Event on FlatList items not working during setInterval()

### ISSUE THREAD

**Issue by malonguwa** [archriss/react-native-snap-carousel#208]
I have a horizontal FlatList, where each time it reaches the end, it automatically adds new elements to the list, so it kind of is an infinite list. I want the app to scroll through the list by itself automatically, but during my testing, when I use timer to call FlatList's scrollToOffset() and make the FlatList auto scrolling, I can not click the item (onPress={this.clickItem} never get called), I guess it has something to do with the time set for setInterval(func, time) because when I increase it to a high value it works (but of course the scrolling is not nice).

May I ask whats the correct way to implement such function? 

Any help would be much appreciated!

Timer function:
```
      this.timer = setInterval(() => {
          console.log('I do not leak!');
          this.flatListRef.scrollToOffset({ offset: this.scrollX+5});
      }, 500);

```
FlatList and its Items:

```
<FlatList
              ref={(ref) => { this.flatListRef = ref; }}
              horizontal = {true}
              data={this.state.dataForCarousel}
              renderItem={this._renderItem}
              onScroll={this.handleScroll}
>
</FlatList>

```
```
  _renderItem = (item) => {
    let item1 = item;
    var txt = item1.item;
    console.log(txt);
    var bgColor = item1.index % 2 == 0 ? 'red' : 'yellow';
    return (
        <TouchableOpacity
            onPress={this.clickItem}
        >
          <View
              style={{ flex: 1, alignItems: "center", justifyContent: "center",backgroundColor: bgColor, width: 80 }}>
            <Text>{txt}</Text>
          </View>
        </TouchableOpacity>
    )
  }

```

**Comment by bd-arc** [comment]
Hi @malonguwa,

I must warn you that you did not follow [the contributing guidelines](https://github.com/archriss/react-native-snap-carousel/blob/master/CONTRIBUTING.md) and that your question has nothing to do with the plugin! It would be better suited in [React Native's dedicated issues](https://github.com/facebook/react-native/issues), and you'll have a much wider audience. Still, since you're being very polite about it, I'll make an effort.

I don't see any problem with your code that would explain the issue. If you create a new app using https://snack.expo.io/ and make sure that the issue can be reproduced in it, I'll gladly take a look at it.

I'm going to close the issue since it has nothing to do with the plugin, but I'll receive your comments and make sure to respond.

**Comment by malonguwa** [comment]
Please accept my apology, I am really new to the Rect Native~ and many thanks for your help and time!

I am willing to implement a horizontal FlatList for stock ticker like in the gif.
![example](https://user-images.githubusercontent.com/2183780/32637352-4073acec-c5f4-11e7-8cb2-92f1b429075a.gif)

So I start to write a really simple App, by using FlatList, and apply a timer to increase FlatList's contentOffsetX, and the FlatList is auto scrolling, but when timer is firing and during the call of 
contentOffsetX increase function, all the FlatList Items' onPress event can not be triggered anymore, I searched a lot about this issue, I see some other people have the same issue... but I not sure if there is something wrong with my implementation to achieve such function. 

Please have a look the demo I put on the snack~

https://snack.expo.io/@lma/flatlistautoscroll

Thanks  again!

**Comment by bd-arc** [comment]
Hi @malonguwa,

I'm currently taking a look at your problem. First of all, note that you should wrap your item's `<View />` container inside the `<Touchable />`, not the other way around. Only what's in the `Touchable` component responds to the `touch` event. This means that only the small `<Text />` can be clicked in your example.

Here is an updated version that allows clicking the entire item: https://snack.expo.io/S1Xcalv1f Note that this doesn't solves your issue yet, but I thought it would still be good to know ;-)

**Comment by bd-arc** [comment]
@malonguwa Unfortunately, I haven't been able to make it work. The inner logic of the `FlatList` component seems to be that when programmatically scrolled, touch events are dismissed.

I've tried [this workaround](https://github.com/facebook/react-native/issues/10822#issuecomment-268142915), but it doesn't seem to work anymore. I've also tried playing with [the gesture responder system](http://facebook.github.io/react-native/docs/gesture-responder-system.html), to no avail.

Here is the updated example: https://snack.expo.io/rkoueGwyG It features my attempt at making it work with a `<ScrollView />`, the underlying component beneath the `FlatList` one. You might be interested in a few things here and there, for example how I made the autoplay stops when reaching the last item.

Make sure to keep me posted if you manage to find a solution ;-)

**Comment by malonguwa** [comment]
@bd-arc Thanks for your time and effort! Really appreciate it! 
I think I make it work :)

After I dump the RN’s own touchable component and implement my own touch event handler.
I also set scrollToOffset animated prop to false and disabled timer in onResponderGrant.

```
        this._gestureHandlers = {
            onStartShouldSetResponder: () => true,  //Allow register response to touch event
            onMoveShouldSetResponder: ()=> true,  //Allow register response to move (scroll) event

            onResponderGrant: ()=> {
                console.log("parent onResponseGrant");
                clearInterval(this.timer);
            }, //when the touch event start (the moment your finger touched the screen)
            onResponderMove: ()=> {
                console.log("parent onResponderMove");
            },  //when the move event start (in here when we scroll the FlatList)

            onResponderRelease: ()=> {
                this.clickItem()
            }, //when we release the touch event (we can think this is the final callBack for the event)

            // onResponderTerminationRequest: () => true
        }

```
I will try to implement the infinite auto scroll at the next step :)

Please refer to https://snack.expo.io/@lma/timerwithcarousel for the demo

**Comment by LuisPChavez** [comment]
Hey @malonguwa I was having trouble with the same thing and I found a workaround but its not the prettiest. In your onPress write a fat arrow function instead of the name of the function you want to execute. For example it should look like this:

onPress{() => {console.log("sdf")}} instead of 

logIfPressed() {
     console.log("sdf")
}

onPress{this.logIfPressed}

Of course this is messy but its the only workaround I found. Good luck!
